.teamcity/settings.kts

```diff
@@ -650,16 +649,179 @@ object WpCalypso : GitVcsRoot({
 		cleanCheckout = true
 	}
 
-	steps {	}
+	steps {
+		script {
+			name = "Prepare environment"
+			scriptContent = """
+				set -e
+
+				export HOME="/calypso"
+				export CHROMEDRIVER_SKIP_DOWNLOAD=true
+				export PUPPETEER_SKIP_DOWNLOAD=true
+				export npm_config_cache=${'$'}(yarn cache dir)
+				export CONFIG_ENV="release"
+
+				# Update node
+				. "${'$'}NVM_DIR/nvm.sh" --no-use
+				nvm install
+
+				# Restore mtime to maximize cache hits
+				/usr/lib/git-core/git-restore-mtime --force --commit-time --skip-missing
+
+				# Decript certs
+				openssl aes-256-cbc -md md5 -d -in desktop/resource/calypso/secrets.json.enc -out config/secrets.json -k "%CALYPSO_SECRETS_ENCRYPTION_KEY%"
+
+				# Install modules
+				yarn install
+			""".trimIndent()
+			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+			dockerPull = true
+			dockerImage = "%docker_image_dekstop%"
+			dockerRunParameters = "-u %env.UID%"
+		}
+
+		script {
+			name = "Build Calypso source"
+			scriptContent = """
+				set -e
+				export HOME="/calypso"
+				export CONFIG_ENV="release"
+
+				# Update node
+				. "${'$'}NVM_DIR/nvm.sh" --no-use
+				nvm install
+
+				# Build desktop
+				yarn run build-desktop:source
+			""".trimIndent()
+			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+			dockerPull = true
+			dockerImage = "%docker_image_dekstop%"
+			dockerRunParameters = "-u %env.UID%"
+		}
+
+		script {
+			name = "Build app (linux)"
+			scriptContent = """
+				set -e
+				export HOME="/calypso"
+				export CONFIG_ENV="release"
+				export ELECTRON_BUILDER_ARGS='-c.linux.target=dir'
+				export USE_HARD_LINKS=false
+
+				# Update node
+				. "${'$'}NVM_DIR/nvm.sh" --no-use
+				nvm install
+
+				# Install app deps
+				yarn run build-desktop:install-app-deps
+
+				# Build app
+				yarn run build-desktop:app
+			""".trimIndent()
+			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+			dockerPull = true
+			dockerImage = "%docker_image_dekstop%"
+			dockerRunParameters = "-u %env.UID%"
+		}
+
+		script {
+			name = "Run tests (linux)"
+			scriptContent = """
+				set -e
+				export HOME="/calypso"
+				export E2EGUTENBERGUSER="%E2EGUTENBERGUSER%"
+				export E2EPASSWORD="%E2EPASSWORD%"
+				export DISPLAY=:99
+				export MOCHA_FILE="./desktop/e2e/result.xml"
+
+				# Update node
+				. "${'$'}NVM_DIR/nvm.sh" --no-use
+				nvm install
+
+				# Start framebuffer
+				Xvfb ${'$'}{DISPLAY} -screen 0 1280x1024x24 &
+
+				# Run tests
+				yarn run test-desktop:e2e
+			""".trimIndent()
+			dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+			dockerPull = true
+			dockerImage = "%docker_image_dekstop%"
+			dockerRunParameters = "-u %env.UID%"
+		}
+	}
+
+	failureConditions {
+		executionTimeoutMin = 15
+	}
+
+	features {
+		feature {
+			type = "xml-report-plugin"
+			param("xmlReportParsing.reportType", "junit")
+			param("xmlReportParsing.reportDirs", "desktop/e2e/result.xml")
+		}
+		perfmon {
+		}
+		pullRequests {
```

Comment:
Expose Pull Requests as branches. It will allow to _manually_ trigger a build for PRs instead of branches (very useful if someone creates a PR from a fork)

Reply:
Right now Circle CI for Desktop bombs out for jobs triggered by forks, so this is a nice value-add!